data-jdbc/src/main/java/io/micronaut/data/jdbc/operations/DefaultJdbcRepositoryOperations.java

```diff
@@ -320,9 +316,10 @@ public <T, R> R findOne(@NonNull PreparedQuery<T, R> pq) {
                                 return o;
                             }
                         };
-                        if (jsonDualityView) {
-                            String viewColumn = getJsonDualityViewColumn(persistentEntity.getAnnotationMetadata());
-                            return readJsonDualityViewEntity(rs, viewColumn, persistentEntity, loadListener, resultType);
+                        QueryResultTransformerInfo queryResultTransformerInfo = preparedQuery.getQueryResultTransformerInfo();
+                        if (queryResultTransformerInfo != null) {
+                            MediaTypeCodec codec = getMediaTypeCodec(queryResultTransformerInfo.getMediaType());
```

Comment:
Can you eliminate the need for `MediaTypeCodec` and use Micronaut Serialization directly? We have an another issue for that

Reply:
You mean inject Micronaut `JsonMapper`? Can you point me to another issue so I can see what needs to be done there.

Reply:
I have added ObjectMapper instead of MediaTypeCodec here and elsewhere where we read JSON from db columns.